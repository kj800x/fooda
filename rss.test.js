const { buildRss } = require("./rss");

describe("rss", () => {
  it("works happy path", () => {
    expect(
      buildRss({
        currentDay: 27,
        menuDay: 27,
        popups: [
          {
            link:
              "http://app.fooda.com/accounts/1729/popup/menu_page/P0733968/items?filterable%5Bvendor_id%5D=6563",
            name: "Jersey Mike's"
          },
          {
            link:
              "http://app.fooda.com/accounts/1729/popup/menu_page/P0733968/items?filterable%5Bvendor_id%5D=4785",
            name: "Brothers Restaurant Tacos"
          }
        ]
      })
    ).toMatchSnapshot();
  });

  it("works missing link", () => {
    expect(
      buildRss({
        currentDay: 27,
        menuDay: 27,
        popups: [
          {
            name: "Jersey Mike's"
          },
          {
            name: "Brothers Restaurant Tacos"
          }
        ]
      })
    ).toMatchSnapshot();
  });
  it("works just one popup", () => {
    expect(
      buildRss({
        currentDay: 27,
        menuDay: 27,
        popups: [
          {
            link:
              "http://app.fooda.com/accounts/1729/popup/menu_page/P0733968/items?filterable%5Bvendor_id%5D=4785",
            name: "Brothers Restaurant Tacos"
          }
        ]
      })
    ).toMatchSnapshot();
  });
  it("works no popups", () => {
    expect(
      buildRss({
        currentDay: 27,
        menuDay: 27,
        popups: []
      })
    ).toMatchSnapshot();
  });
  it("works menu not today", () => {
    expect(
      buildRss({
        currentDay: 25,
        menuDay: 27,
        popups: [
          {
            link:
              "http://app.fooda.com/accounts/1729/popup/menu_page/P0733968/items?filterable%5Bvendor_id%5D=6563",
            name: "Jersey Mike's"
          },
          {
            link:
              "http://app.fooda.com/accounts/1729/popup/menu_page/P0733968/items?filterable%5Bvendor_id%5D=4785",
            name: "Brothers Restaurant Tacos"
          }
        ]
      })
    ).toMatchSnapshot();
  });
});
