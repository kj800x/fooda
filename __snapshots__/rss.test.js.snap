// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rss works happy path 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>
<rss version=\\"2.0\\">
<channel>
 <title>Davenport Fooda</title>
 <description>Kevin Johnson's Davenport Fooda RSS feed</description>
 <link>https://fooda.com/my</link>
 <lastBuildDate>Thu, 27 Feb 2020 22:46:39 GMT</lastBuildDate>
 <pubDate>Thu, 27 Feb 2020 11:00:00 GMT</pubDate>
 <ttl>10</ttl>

 
</channel>
</rss>"
`;

exports[`rss works just one popup 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>
<rss version=\\"2.0\\">
<channel>
 <title>Davenport Fooda</title>
 <description>Kevin Johnson's Davenport Fooda RSS feed</description>
 <link>https://fooda.com/my</link>
 <lastBuildDate>Thu, 27 Feb 2020 22:46:39 GMT</lastBuildDate>
 <pubDate>Thu, 27 Feb 2020 11:00:00 GMT</pubDate>
 <ttl>10</ttl>

 
</channel>
</rss>"
`;

exports[`rss works menu not today 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>
<rss version=\\"2.0\\">
<channel>
 <title>Davenport Fooda</title>
 <description>Kevin Johnson's Davenport Fooda RSS feed</description>
 <link>https://fooda.com/my</link>
 <lastBuildDate>Thu, 27 Feb 2020 22:46:39 GMT</lastBuildDate>
 <pubDate>Thu, 27 Feb 2020 11:00:00 GMT</pubDate>
 <ttl>10</ttl>

 
 <item>
 <title>
   Thursday at Davenport: Jersey Mike's
 </title>
 <description></description>
 <link>http://app.fooda.com/accounts/1729/popup/menu_page/P0733968/items?filterable%5Bvendor_id%5D=6563</link>
 <pubDate>Thu, 27 Feb 2020 11:00:00 GMT</pubDate>
</item>
 

 <item>
 <title>
   Thursday at Davenport: Brothers Restaurant Tacos
 </title>
 <description></description>
 <link>http://app.fooda.com/accounts/1729/popup/menu_page/P0733968/items?filterable%5Bvendor_id%5D=4785</link>
 <pubDate>Thu, 27 Feb 2020 11:00:00 GMT</pubDate>
</item>
 
</channel>
</rss>"
`;

exports[`rss works missing link 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>
<rss version=\\"2.0\\">
<channel>
 <title>Davenport Fooda</title>
 <description>Kevin Johnson's Davenport Fooda RSS feed</description>
 <link>https://fooda.com/my</link>
 <lastBuildDate>Thu, 27 Feb 2020 22:46:39 GMT</lastBuildDate>
 <pubDate>Thu, 27 Feb 2020 11:00:00 GMT</pubDate>
 <ttl>10</ttl>

 
</channel>
</rss>"
`;

exports[`rss works no popups 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>
<rss version=\\"2.0\\">
<channel>
 <title>Davenport Fooda</title>
 <description>Kevin Johnson's Davenport Fooda RSS feed</description>
 <link>https://fooda.com/my</link>
 <lastBuildDate>Thu, 27 Feb 2020 22:46:39 GMT</lastBuildDate>
 <pubDate>Thu, 27 Feb 2020 11:00:00 GMT</pubDate>
 <ttl>10</ttl>

 
</channel>
</rss>"
`;
